source/includes/inkling-reference/_lessons.html.md

```diff
@@ -88,100 +84,269 @@ end
+
+In this example we show how configuration works for `bricks_percent`.
+
+The instructor selects values for `bricks_percent` from the given range. `Float32 {0.1:0.01:1.0}` is an Inkling specification for a constrained type. In a constrained type the values are all `Float32` but they also obey the constraint specified.
+
+We can also configure `bricks_percent` to take a constant value:
+
+ `constrain bricks_percent with Float32 {1.0}`
+
+`Float32 {1.0}` is a Float32 constrained to take the value 1.0. This is the
+value list form of the range expression. You can add values to the list as long as they are of the same type, so the following is also valid:
+
+`constrain bricks_percent with Float32 {1.0, 1.5}`
+
+###### Train and Test Clause
+
+The `test` and `train` clauses describe testing and training.
+
+The `from` subclause in the test/train syntax is used to name, describe, and select the
+training data that is sent by the simulator to the lesson.
+
+The `test` clause and the `train` clause have identical syntax except for
+their keyword (`train` or `test`).  
+
+> Train and Test Clause Syntax
 
 ```inkling--syntax
 trainClause ::=
 train
-  fromClause
-  send <name>
-trainingSpecifer
-```
+  from <item_name> in <simulator_name>
+  select <item_name>
+  send <item_name>
 
-```inkling--syntax
 testClause ::=
 test
-  fromClause
-  send <name>
-trainingSpecifer
+  from <item_name> in <simulator_name>
+  select <item_name>
+  send <item_name>
 ```
 
-[//]: # (  [expect <name>]?    # only valid for data or generator)
+### Usage
+
+The `test` and `train` clauses are optional. 
 
+If neither the `test` or `train` lesson clauses are present, defaults for both
+clauses are generated. The default in both cases is:
 
-Select the Syntax tab for the syntax for these clauses.
+`from item in <simulator_name> select item send item`
 
-The **test** clause and the **train** clause have identical syntax except for
-their keyword (**train** or **test**).  However they both vary depending on the
-_trainingSpecifier_ in the curriculum. 
+(Here `<simulator_name>` refers to the name of the simulator being used by the
+lesson.)
 
-[//]: # (Note that the **expect** is only available in those cases that have known expected values, and that occurs when the _trainingSpecifier_ is **data** or **generator**.)
+If one of the `train` and `test` clauses is present, no defaults are generated.
 
-The **from** clause in the test/train syntax is used to name and describe the
-training data that is sent by the system (either from a labeled data set, in the
-**data** case, or by the generator or simulator) to the lesson.
+If the `train` clause is not present, the return schema of the simulator must exactly match the input schema to the network.
 
-###### Lesson Until Clause Syntax
+The `test` clause is not required for any particular lesson. But if the final lesson does not have a `test` clause that is an error.
+
+### Example
+
+In this example we show `train` and `test` clauses. 
```

Comment:
Shouldn't we compress this down into only showing the relevant information here? Could we only show the curriculum?